packages/grafana-ui/src/components/Switch/Switch.tsx

```diff
@@ -104,6 +99,24 @@ const getSwitchStyles = (theme: GrafanaTheme2, transparent?: boolean) => ({
         },
       },
 
+      '&:disabled + label': {
+        background: theme.colors.action.disabledBackground,
+        borderColor: theme.colors.border.weak,
+        color: theme.colors.text.disabled,
+        cursor: 'not-allowed',
+
+        '&:hover': {
+          background: theme.colors.action.disabledBackground,
+        },
+      },
+
+      '&:disabled:checked + label': {
+        '&::after': {
+          transform: 'translate3d(18px, -50%, 0)',
```

Comment:
Don't think this is necessary for this case right? It should already be set in line 97

Reply:
you're right!